lib/entity/generate_transactions_button.dart

```diff
@@ -171,42 +172,32 @@ addPeriodicDataToList(
         random.nextInt(configData['maxAmount'] - configData['minAmount']);
     listData.add({
       'amount': amount,
-      'ben_name': "Beneficiary",
+      'ben_name': generateName(),
       'reference': "Example Transaction",
       'rem_name': configData['benName'],
       'Type': configData['credit'] ? "Credit" : "Debit",
       'timestamp': configDate,
       'day': "${dateIterator.format(DATE_FORMAT)}/${dateIterator.EEEE}",
     });
   });
+
   return listData;
 }
 
+addPeriodicDataToList(
+    List<Map<String, dynamic>> configList, Jiffy dateIterator, int Counter) {
+  var configData = configList.where((ele) => ele['day'] == Counter).toList();
+
+  return generateTransactionData(configData, dateIterator);
+}
+
 addRandomDataToList(List<Map<String, dynamic>> configList, String period,
     Jiffy dateIterator, int Counter) {
   List<Map<String, dynamic>> listData = [];
-  var random = math.Random();
+
   configList.forEach((configData) {
     configData[period].where((date) => date == Counter).forEach((config) {
-      DateTime configDate = DateTime(
-          dateIterator.year,
-          dateIterator.month,
-          dateIterator.date,
-          random.nextInt(24),
-          random.nextInt(60),
-          random.nextInt(3600),
-          random.nextInt(3600000));
-      double amount = configData['minAmount'] +
-          random.nextInt(configData['maxAmount'] - configData['minAmount']);
-      listData.add({
-        'amount': amount,
```

Comment:
Don't delete any code in generate transaction

Reply:
this is a refactor because this exact code appeared twice

Reply:
just pulled from main, i'll fix the implementation to adapt to it